swap_meet/vendor.py

```diff
@@ -1,2 +1,62 @@
 class Vendor:
-    pass
+    def __init__(self, inventory = None):
+        self.inventory = inventory if inventory is not None else []
+
+    def add(self, inventory_item):
+        self.inventory.append(inventory_item)
+        return inventory_item
+
+    def remove(self, inventory_item):
+        if inventory_item in self.inventory:
+            self.inventory.remove(inventory_item)
+            return inventory_item
+        else:
+            return False
+
+    # inventory is a list of item istances with each item having a category as an attribute
+    # get_by_category filters inventory by category
+    # items is a list of just the filtered item instances
+    def get_by_category(self, category):
+        items = []
+        for item in self.inventory:
+            if item.category == category:
+                items.append(item)
+        return items
+
+    def swap_items(self, vendor, my_item, their_item):
+        if my_item not in self.inventory:
+            return False
+        if their_item not in vendor.inventory:
+            return False
+        self.remove(my_item)
+        vendor.add(my_item)
+        vendor.remove(their_item)
+        self.add(their_item)
+        return True
+        
+    def swap_first_item(self, vendor):
+        if not self.inventory or not vendor.inventory:
+            return False
+        vendor.add(self.inventory[0])
+        self.remove(self.inventory[0])
+        self.add(vendor.inventory[0])
+        vendor.remove(vendor.inventory[0])
+        return True
```

Comment:
Nice!! This is a really clean solution

Reply:
We can also assign multiple variables in one line like so:

```
        self.inventory[0], friend_vendor.inventory[0] = friend_vendor.inventory[0], self.inventory[0]
```        

Here's more info on Python multiple assignment: https://www.w3schools.com/python/gloss_python_assign_value_to_multiple_variables.asp